flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSubpartitionMemoryDataManager.java

```diff
@@ -247,6 +261,17 @@ public void releaseSubpartitionBuffers(List<BufferIndexAndChannel> toRelease) {
                                 }));
     }
 
+    /** Un-synchronized get number of buffers not consumed. */
+    @SuppressWarnings("FieldAccessNotGuarded")
+    public int getNumOfBuffersUnsafe() {
+        return unConsumedBuffers.size();
+    }
+
+    /** Un-synchronized get number of bytes not consumed. */
+    public long getNumOfBytesUnsafe() {
+        return getNumOfBuffersUnsafe() * (long) bufferSize;
+    }
+
```

Comment:
I'm not entirely sure about these calculations. What are the semantics here?

Reply:
It is used for provide information of queued buffers. After some consideration, I think it is of little significance to provide this information for non pipelined shuffle, so I removed this part of the calculation now.